src/WidgetBase.ts

```diff
@@ -160,6 +180,44 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 				propertiesChangedListeners[i].call(this, evt);
 			}
 		}));
+
+		// Add a render decorator that will register callbacks for the v-node 'afterCreate' and
+		// 'afterUpdate' lifecycle methods to call the widget lifecycle methods onElementCreated
+		// and onElementUpdated
+		this.addDecorator('afterRender', (node: DNode) => {
+			// Using callback functions that were bound outside of this render call to make sure
+			// the functions don't change.
+			decorate(node, (node: HNode) => {
+				node.properties.afterCreate = afterCreateCallback;
+			}, isHNodeWithKey);
+
+			decorate(node, (node: HNode) => {
+				node.properties.afterUpdate = afterUpdateCallback;
```

Comment:
could the after update hook be added in the decorator call above? saves having to iterate the nodes twice?